### PR TITLE
update validation columns for get_sce

### PR DESCRIPTION
**Pull request by myushen**
separate the column validation from `get_sce` and `get_pseudobulk`

### CODE REVIEW

**Review by stemangiola — CHANGES_REQUESTED** [review]
here we are overengineering, and creating complex general functions, that should not be.

The validation in this case should be in get_single_Cell_experiment function.

I don't think `param_validation ` should exist, you can add the context of the validation in the `get_single_Cell_experiment` directly

**Comment by myushen** [comment]
separating `param_validation` from the main function can minimize a few lines of code duplication for `get_sce` and `get_pseudobulk`  I think

**Comment by stemangiola** [comment]
> separating `param_validation` from the main function can minimize a few lines of code duplication for `get_sce` and `get_pseudobulk` I think

Agree, but in this case this minimidsation has a big cost of complexity, and code clarity. In our case `get_single_Cell_experiment ` if basically empty, so a validation header is acceptable.

File: R/counts.R [hunk 1]
```diff
@@ -94,15 +109,42 @@ get_summarized_experiment <- function(
     features = NULL,
     type = "sce"
 ) {
-  repository <- if (type == "sce") COUNTS_URL else pseudobulk_url
```
Comment (stemangiola) [inline R/counts.R:99]: I propose this function.

```r
get_data_container <- function(
    data,
    assays = "counts",
    cache_directory = get_default_cache_dir(),
    repository, # url
    grouping_column, # file_id_db, file_id
    features = NULL
)
```

**Comment by stemangiola** [comment]
Thanks @myushen !

**Pull request merged**